### PR TITLE
Support Amazon Q inline suggestion in remote dev

### PR DESCRIPTION
**Pull request by andrewyuq**
1. Use InlineSuffixRenderer in remote dev to enable remote host inlay rendering.
2. Conditionally position the popup in remote dev env due to a different JetBrains' implementation of the popup in remote(`BackendBeAbstractPopup`) and its different ways to handle showing in specific positions than the ways the local version `AbstractPopup` do.
3. Display info in various places to inform users that Amazon Q inline(CodeWhisperer) will only be supported in JB remote dev IDE version 2023.3+(inclusive)
4. Other UX fixes for inline suggestions UX flow

Known issues:
1. popup will not change position according to other suggestions (from the 2nd and so on) once shown. it's a limitation of the current API used.
2. Kebab menu in the popup is temporary disabled due to not properly being displayed.

https://github.com/aws/aws-toolkit-jetbrains/assets/89420755/385d6292-b08e-4f18-b13d-aae60568edcc

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/inlay/CodeWhispererInlayManager.kt [hunk 1]
```diff
@@ -52,19 +65,36 @@ class CodeWhispererInlayManager {
...
+            ).forEach { disposable ->
+                Disposer.dispose(disposable)
+            }
+            editor.inlayModel.getBlockElementsInRange(
+                0,
+                editor.document.textLength,
+                InlineBlockElementRenderer::class.java
+            ).forEach { disposable ->
+                Disposer.dispose(disposable)
+            }
```
Comment (rli) [inline plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/inlay/CodeWhispererInlayManager.kt:97]:
```suggestion
        val inlineRenderer: Class<out EditorCustomElementRenderer>
        val blockRenderer: Class<out EditorCustomElementRenderer>

        if (!AppMode.isRemoteDevHost()) {
            inlineRenderer = CodeWhispererInlayInlineRenderer::class.java
            blockRenderer = CodeWhispererInlayBlockRenderer::class.java
        } else {
            inlineRenderer = InlineSuffixRenderer::class.java
            blockRenderer = InlineBlockElementRenderer::class.java
        }

        editor.inlayModel.getInlineElementsInRange(
            0,
            editor.document.textLength,
            inlineRenderer
        ).forEach { disposable ->
            Disposer.dispose(disposable)
        }
        editor.inlayModel.getBlockElementsInRange(
            0,
            editor.document.textLength,
            blockRenderer
        ).forEach { disposable ->
            Disposer.dispose(disposable)
        }
```

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/inlay/CodeWhispererInlayManager.kt [hunk 2]
```diff
@@ -52,19 +65,36 @@ class CodeWhispererInlayManager {
...
+                0,
+                editor.document.textLength,
+                InlineSuffixRenderer::class.java
+            ).forEach { disposable ->
+                Disposer.dispose(disposable)
+            }
+            editor.inlayModel.getBlockElementsInRange(
+                0,
+                editor.document.textLength,
+                InlineBlockElementRenderer::class.java
```
Comment (rli) [inline plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/inlay/CodeWhispererInlayManager.kt:94]: do we need to make sure these were all from codewhisperer

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/utils/RemoteEnvUtils.kt [hunk 3]
```diff
@@ -21,3 +24,8 @@ fun disableExtensionIfRemoteBackend() {
         throw ExtensionNotApplicableException.create()
     }
 }
+
+// CW can be supported only after at least build 232.9921.47 on remote env
+fun isRunningOnCWNotSupportedRemoteBackend() =
+    ApplicationInfo.getInstance().build.compareTo(BuildNumber.fromStringOrNull("232.9921.47")) < 0 &&
+        AppMode.isRemoteDevHost()
```
Comment (rli) [inline plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/utils/RemoteEnvUtils.kt:31]: can you change `isRunningOnRemoteBackend` to use `AppMode.isRemoteDevHost()`

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/inlay/CodeWhispererInlayManager.kt (hunk 2 above)
Comment (andrewyuq) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/inlay/CodeWhispererInlayManager.kt:94]: we should, haven't figured out how, let me think a bit.

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/utils/RemoteEnvUtils.kt (hunk 3 above)
Comment (andrewyuq) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/utils/RemoteEnvUtils.kt:31]: sure

**Review by rli — APPROVED** [review]

**Pull request merged**